### PR TITLE
feat(nms): Clean up directory organization

### LINKED ISSUE #11890: [NMS] Clean up directory organization
# Clean up directory organization of `/nms/packages/magmalte`

## Summary
This is a task to migrate everything from `magma/nms/packages/magmalte` to `magma/nms`. Some of the basic testing and building workflows present some difficulty here because of the doubled `package.json` files in those two directories.

Some other dependencies on the current structure that need to be cleaned up:
- docker-compose files
- .circleci files
- .md documentation files
- .yaml deployment files

### PR DESCRIPTION
**Pull request by thmsschmitt**
## Summary
Migrate everything from `magma/nms/packages/magmalte` to `magma/nms`. Fixes #11890

## Tests done locally

- run yarn test
- run yarn eslint
- run yarn flow
- rebuild orc8r docker containers
- rebuild nms docker containers
- basic smoke testing using the app
- run e2e tests (e2e_test_setup.sh)

### CODE REVIEW

File: docs/docusaurus/versioned_docs/version-1.5.X/proposals/p010_subscriber_scaling.md
Comment (Neudrino) [inline docs/docusaurus/versioned_docs/version-1.5.X/proposals/p010_subscriber_scaling.md]: This special change in path away from `master` to a version in the documentation of version 1.5 is good, however, we should point to the `v1.5` branch.

There is only one opening bracket `(` needed, afaiks.

File: docs/docusaurus/versioned_docs/version-1.5.X/nms/architecture_overview.md
Comment (Neudrino) [inline docs/docusaurus/versioned_docs/version-1.5.X/nms/architecture_overview.md]: We are only moving the content on master, so we should generally not touch the documentation for version 1.5.

File: docs/docusaurus/versioned_docs/version-1.5.X/orc8r/upgrade_1_5.md
Comment (Neudrino) [inline docs/docusaurus/versioned_docs/version-1.5.X/orc8r/upgrade_1_5.md]: We are only moving the content on master, so we should generally not touch the documentation for version 1.5.

File: docs/docusaurus/versioned_docs/version-1.6.X/orc8r/upgrade_1_5.md
Comment (Neudrino) [inline docs/docusaurus/versioned_docs/version-1.6.X/orc8r/upgrade_1_5.md]: As above: Documentation for version 1.6 should probably not be changed.

File: docs/docusaurus/versioned_docs/version-1.7.0/nms/architecture_overview.md
Comment (Neudrino) [inline docs/docusaurus/versioned_docs/version-1.7.0/nms/architecture_overview.md]: As above: Documentation for version 1.7 should probably not be changed.

File: docs/docusaurus/versioned_docs/version-1.7.0/orc8r/upgrade_1_5.md
Comment (Neudrino) [inline docs/docusaurus/versioned_docs/version-1.7.0/orc8r/upgrade_1_5.md]: As above: Documentation for version 1.7 should probably not be changed.

File: docs/docusaurus/versioned_docs/version-1.7.0/proposals/p010_subscriber_scaling.md
Comment (Neudrino) [inline docs/docusaurus/versioned_docs/version-1.7.0/proposals/p010_subscriber_scaling.md]: This is a good change. Thanks!

File: docs/readmes/orc8r/upgrade_1_5.md
Comment (Neudrino) [inline docs/readmes/orc8r/upgrade_1_5.md]: For the upgrade to v1.5 it should probably not be changed.

File: docs/readmes/proposals/p010_subscriber_scaling.md
Comment (Neudrino) [inline docs/readmes/proposals/p010_subscriber_scaling.md]: This is a proposal on master, so we should probably stay with branch master and update the path to reflect the actual move target.

File: nms/Makefile
Comment (Neudrino) [inline nms/Makefile:3]: Was this file migrated (does not look like it)?
If it was newly introduced, we might want to consider syncing with the Bazel effort or evaluate other options to not newly introduce make for yet another component. :man_shrugging:

File: nms/README.md
Comment (Neudrino) [inline nms/README.md]: In perspective of view from the root of the repository, the path would be `nms/` not `magma/nms`. (This applies for all other references below as well...)
To make this more clear, it can also link to itself, e.g. the following might work:
```markdown
[`nms/`](../nms/)
```
If you really want to have the magma repository root specified, you might want to consider `${MAGMA_ROOT}` instead.

File: nms/docker-compose.yml
Comment (Neudrino) [inline nms/docker-compose.yml:25]: I wonder, if this needs to be specified, or if the current directory is the default option, when non is specified?

File: nms/scripts/fuji-upgrade/runs-on-nms.sh
Comment (Neudrino) [inline nms/scripts/fuji-upgrade/runs-on-nms.sh:65]: With https://github.com/magma/magma/pull/12437/files#diff-58ffecb65a2cd39c9c5318019618d288188704774538df4ce30820f63864bc9aR2, does `fbc` need to be changed here as well?

**Review by Neudrino — COMMENTED** [review]
Reviewed mainly the `docs` part of the PR.

File: docs/docusaurus/versioned_docs/version-1.5.X/orc8r/upgrade_1_5.md
Comment (thmsschmitt) [reply docs/docusaurus/versioned_docs/version-1.5.X/orc8r/upgrade_1_5.md]: And leave behind broken links?

Comment (thmsschmitt) [reply docs/docusaurus/versioned_docs/version-1.5.X/orc8r/upgrade_1_5.md]: I think the point here is that people should be able to run the migration script and we should probably point to the latest version in case we want to fix something.

File: nms/Makefile
Comment (thmsschmitt) [reply nms/Makefile:3]: Yes it was https://github.com/magma/magma/pull/12437/files/b4ace80145a1d9c57f671b53af436eb80caea0f6#diff-e179ebbfce7456303ff88b391aca4990f357d44e66ef0e6497e4b3227a15455c

File: nms/scripts/fuji-upgrade/runs-on-nms.sh
Comment (thmsschmitt) [inline nms/scripts/fuji-upgrade/runs-on-nms.sh]: This is incorrect it would be executed on the old version.

File: docs/readmes/proposals/p010_subscriber_scaling.md
Comment (thmsschmitt) [reply docs/readmes/proposals/p010_subscriber_scaling.md]: It was a proposal for v1.5 see line 523. To make this stable I will also change this to `v1.5` if I find the corresponding code.

**Review by themarwhal — APPROVED** [review]
Orc8r and doc chagnes lgtm!

File: nms/scripts/fuji-upgrade/runs-on-nms.sh
Comment (thmsschmitt) [reply nms/scripts/fuji-upgrade/runs-on-nms.sh:65]: This refers to the archived [fbcnms packages](https://github.com/magma/fbc-js-core), but shouldn't be changed because it runs on an old version.

File: nms/docker-compose.yml
Comment (thmsschmitt) [reply nms/docker-compose.yml:25]: I just tried it, looks like it is required: 
> Service magmalte has neither an image nor a build context specified. At least one must be provided.

File: docs/docusaurus/versioned_docs/version-1.5.X/nms/architecture_overview.md
Comment (andreilee) [inline docs/docusaurus/versioned_docs/version-1.5.X/nms/architecture_overview.md]: Version 1.5 docs should refer to the v1.5 branch of the codebase, so this change shouldn't be included in this versioned doc.

File: docs/docusaurus/versioned_docs/version-1.6.X/orc8r/upgrade_1_5.md
Comment (andreilee) [inline docs/docusaurus/versioned_docs/version-1.6.X/orc8r/upgrade_1_5.md:69]: Typo here, it looks like this should be pointing to v1.6 branch

File: docs/docusaurus/versioned_docs/version-1.7.0/orc8r/upgrade_1_5.md
Comment (andreilee) [inline docs/docusaurus/versioned_docs/version-1.7.0/orc8r/upgrade_1_5.md:69]: Typo here, this looks like it should be pointing to v1.7 branch

**Review by andreilee — CHANGES_REQUESTED** [review]
Thanks for putting up this incredibly comprehensive PR!

I noticed a few typo-like errors, and I will also ask that you include in the PR description what tests you ran (eg. bringing up and setting up an NMS, running e2e tests, flow, eslint, unit tests). In case we find a bug later on, it's easier to look at the commit/PR history and see where a bug was introduced.

File: docs/docusaurus/versioned_docs/version-1.6.X/orc8r/upgrade_1_5.md
Comment (thmsschmitt) [reply docs/docusaurus/versioned_docs/version-1.6.X/orc8r/upgrade_1_5.md:69]: That was intentional because this is a script to upgrade to version 1.5. I can change this, but wouldn't that mean we have to adapt the link with every release?

File: docs/docusaurus/versioned_docs/version-1.7.0/orc8r/upgrade_1_5.md
Comment (thmsschmitt) [reply docs/docusaurus/versioned_docs/version-1.7.0/orc8r/upgrade_1_5.md:69]: Same as about, I wasn't sure what the best strategy is here.

**Review by xbend — APPROVED** [review]
dp LGTM

File: docs/docusaurus/versioned_docs/version-1.6.X/orc8r/upgrade_1_5.md
Comment (electronjoe) [reply docs/docusaurus/versioned_docs/version-1.6.X/orc8r/upgrade_1_5.md:69]: @andreilee I think your reply is necessary to unblock

**Comment by nstng** [comment]
CI changes looking good.
* Please rebase - you are on a broken go state. This should fix cloud-workflow, Domain proxy workflow and build-all. Then I would LGTM this.
* Are the security alerts relevant? I guess this is out of scope here because of re-running on moved sources - but should this be tracked? See https://github.com/magma/magma/pull/12437/checks?check_run_id=6077286104

**Comment by thmsschmitt** [comment]
> * Are the security alerts relevant? I guess this is out of scope here because of re-running on moved sources - but should this be tracked?

Yes, they rerun because of the move, no security relevant changes have been made. And yes we should track and tackle them at some point 🙄

**Review by nstng — APPROVED** [review]
CI changes lgtm :+1:

**Review by Neudrino — APPROVED** [review]

**Comment by Neudrino** [comment]
Hi @andreilee , could you have a look at this PR again, please. Your review seems to be blocking the further development on this side.

**Review by andreilee — APPROVED** [review]

File: docs/docusaurus/versioned_docs/version-1.6.X/orc8r/upgrade_1_5.md
Comment (andreilee) [reply docs/docusaurus/versioned_docs/version-1.6.X/orc8r/upgrade_1_5.md:69]: @thmsschmitt hm you're right. My mistake.

File: docs/docusaurus/versioned_docs/version-1.7.0/orc8r/upgrade_1_5.md
Comment (andreilee) [reply docs/docusaurus/versioned_docs/version-1.7.0/orc8r/upgrade_1_5.md:69]: Ignore my comment here. Assume same response here as what I replied with elsewhere.

**Comment by andreilee** [comment]
@Neudrino Looks like we'll need an approval from @magma/approvers-orc8r-infra as well.

**Review by hcgatewood — APPROVED** [review]
Orc8r infra changes lgtm

**Comment by hcgatewood** [comment]
Force-merging to unblock

**Pull request merged**
